src/main/java/org/jenkinsci/test/acceptance/docker/fixtures/GitLabContainer.java

```diff
@@ -119,4 +104,41 @@ public String createUserToken(String userName, String password, String email, St
                 .verifyOrDieWith("Unable to create user")
                 .trim();
     }
+
+    // attempt to cleanup, no retry
+    public void cleanup(String token, String repoName, String groupName) {
+        try (var gitlabapi = new GitLabApi(getHttpUrl(), token)
+                .withRequestTimeout(GITLAB_API_CONNECT_TIMEOUT_MS, GITLAB_API_READ_TIMEOUT_MS)) {
+
+            try {
+                gitlabapi.getProjectApi().getProjects(repoName).stream()
+                        .filter(proj -> repoName.equals(proj.getName()))
+                        .findFirst()
+                        .ifPresent(project -> {
+                            try {
+                                gitlabapi.getProjectApi().deleteProject(project);
+                            } catch (GitLabApiException e) {
+                                throw new RuntimeException("Failed to delete project: " + repoName, e);
+                            }
+                        });
+            } catch (Exception e) {
+                System.err.println("Failed to delete repo '" + repoName + "': " + e.getMessage());
+            }
+
+            try {
+                gitlabapi.getGroupApi().getGroups(groupName).stream()
+                        .filter(g -> groupName.equals(g.getName()))
+                        .findFirst()
+                        .ifPresent(group -> {
+                            try {
+                                gitlabapi.getGroupApi().deleteGroup(group.getId());
+                            } catch (GitLabApiException e) {
+                                throw new RuntimeException("Failed to delete group: " + groupName, e);
+                            }
+                        });
+            } catch (Exception e) {
+                System.err.println("Failed to delete group '" + groupName + "': " + e.getMessage());
```

Comment:
Rethrow as `AssertionErrror`?

Reply:
container cleanup included again, but not it doesn't swallow exception